Add commercial support, fix WASM and issues with Qt 6.7.+

Kidev opened the pull request:
## This PR includes
- Updates to parameters and logic to handle the latest `aqtinstall` versions, making the action once again able to install any version of Qt, WASM and extensions included (`aqtinstall` >=3.2.0)  
- Add full support for installing Qt commercial versions provided you own a Qt account (paid or not). The account is required even if you only download open-source versions (`aqtinstall` >=3.2.2)  
- Updated outdated defaults, and used Qt 6.8.3 as LTS  
- Updated @action/cache to v4 to match Github's update

### Context
Since Qt 6.7+, the Qt team decided to make WASM a part of `all_os` as host, with target `wasm`. This is sensible as the WASM version of Qt cannot work on its own and needs a host. To install Qt WASM for the latest versions, the `host` and `target` values must be edited. See examples.  
This change is required by the latest versions of `aqtinstall` (>=3.2.0)

## Examples:  

> [!TIP]
> Click [run] to see a successful Github Action run using each example
> Click [workflow] to see the corresponding workflow

### Latest Qt WASM [[run](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/14232248777/job/39885109392#step:6:1)] [[workflow](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/14232248777/workflow)]
```yaml
- name: Install Qt for host and WASM
  uses: jurplel/install-qt-action@v4
  with:
    version: '6.8.2'
    host: 'all_os'
    target: 'wasm'
    arch: 'wasm_singlethread'
    modules: 'qtquick3d'
```
Notice `host` and `target`. Since `--autodesktop` is always implied, this will download WASM for the current OS (so this will download `linux_gcc_64` and `wasm_singlethread` if ran on `linux`).  

### Qt commercial [[run](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/14366206584/job/40279631228#step:6:1)] [[workflow](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/14366206584/workflow)]: 
```yaml
- name: Install Qt commercial
  uses: jurplel/install-qt-action@v4
  with:
    version: '6.8.2'
    target: 'desktop'
    arch: 'linux_gcc_64'
    use-official: true
    email: ${{ secrets.QT_USERNAME }}
    pw: ${{ secrets.QT_PASSWORD }}
    modules: 'qtquick3d'
```
Notice the absence of `host`: the official installer only installs using the current OS as `host`. If provided, it will be ignored. You should use secrets for the login information. This will download `linux_gcc_64` for `linux` using the official installer. If you own a paid Qt license, you can use a commercial only version in the `version` field, and it will work.   

### Qt commercial WASM [[run](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/14232259167/job/39885140618#step:6:1)] [[workflow](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/14232259167/workflow)] 
```yaml
- name: Install Qt commercial
  uses: jurplel/install-qt-action@v4
  with:
    version: '6.8.2'
    target: 'desktop'
    arch: 'linux_gcc_64'
    use-official: true
    email: ${{ secrets.QT_USERNAME }}
    pw: ${{ secrets.QT_PASSWORD }}
    modules: 'wasm_singlethread qtquick3d'
```
Same as the previous one, `host` is implied. For the official installer however, there is no `--autodesktop`. Instead, the `modules` field get a widened meaning, and includes all [Qt packages](https://doc.qt.io/qt-6/get-and-install-qt-cli.html#component-names-for-installation). So this example will install Qt for `linux` (implied to be the `host`), for the WASM host architecture `linux_gcc_64`, and will download the `wasm_singlethread` package (considered the target) along with `qtquick3d` for both.

jdpurcell commented:
@Kidev `install-qt-action` already passes `--autodesktop`, it's just that it doesn't typically work if you specify `aqtsource` as you were probably doing for testing. The logic for when it passes in the parameter is determined by:
```ts
const isAutodesktopSupported = async (): Promise<boolean> => {
  const rawOutput = await getPythonOutput("aqt", ["version"]);
  const match = rawOutput.match(/aqtinstall\(aqt\)\s+v(\d+\.\d+\.\d+)/);
  return match ? compareVersions(match[1], ">=", "3.0.0") : false;
};
```
The problem is, if you point `aqtsource` at a branch on your Git repo, due to aqt's use of `setuptools-scm`, the version is auto-generated like `v0.1.dev1823` so the regex doesn't match it due to the letters, and it falls to the `: false` expression of the ternary conditional operator. In my opinion that should be changed to `: true` to allow easier use with `aqtsource`.

Right now as you discovered, one either needs to pass `extra: '--autodesktop'`, or only point `aqtsource` to a version tag that satisfies the version expectation. For example when I was testing some things with `aqt`, I created a fake version tag like `v3.1.100` for use with `aqtsource` so that `setuptools-scm` would auto-assign it that version.

Kidev commented:
Indeed, I'll remove it from the README and such then. In the meantime I got it all working, [as you can see in this demo right here](https://github.com/Kidev/QtQuick3D-Tools/blob/0fc39d219b716a6c1fdc143479f0e2f8ec5fdfdd/.github/workflows/build-deploy-demo.yml#L39). You can use it right now if you need to, no need to wait for either PRs:
```yml
- name: Install Qt for host architecture
  uses: Kidev/install-qt-action@v4.2.0
  with:
    version: '6.8.1'
    host: 'all_os'
    target: 'wasm'
    arch: 'wasm_singlethread'
    set-env: 'true'
    modules: 'qtquick3d'
    aqtsource: 'git+https://github.com/Kidev/aqtinstall.git@wasm'
    extra: '--autodesktop'
```

crueter commented:
@Kidev My application depends on Qt Quick, and this doesn't seem to properly install it for WASM or Android.

```
Failed to find required Qt component "Quick".
```

Here's my config:
```
- name: Install Qt (Host)
  uses: Kidev/install-qt-action@v4.2.0
  with:
    cache: on
    version: 6.8.1

- name: Install Qt (WASM)
  if: // ...
  uses: Kidev/install-qt-action@v4.2.0
  with:
    cache: on
    host: all_os
    target: wasm
    version: 6.8.1
    arch: wasm_singlethread
    set-env: true
    modules: 'qtquick3d'
    aqtsource: 'git+https://github.com/Kidev/aqtinstall.git@wasm'
```

Why is Quick not getting installed?

crueter commented:
Actually, scrolling up, it can't find Qt core...

```
Found package configuration file:

    /.../lib/cmake/Qt6Quick/Qt6QuickConfig.cmake

  but it set Qt6Quick_FOUND to FALSE so package "Qt6Quick" is considered to
  be NOT FOUND.  Reason given by package:

  Qt6Quick could not be found because dependency Qt6Core could not be found.
```

...because it can't find Qt6CoreTools.

```
Found package configuration file:

    /.../lib/cmake/Qt6Core/Qt6CoreConfig.cmake

  but it set Qt6Core_FOUND to FALSE so package "Qt6Core" is considered to be
  NOT FOUND.  Reason given by package:

  Qt6Core could not be found because dependency Qt6CoreTools could not be
  found.
```

crueter commented:
Nevermind, I had the wrong host paths set in my configure command.

Kidev commented:
[The PR fixing WASM and more](https://github.com/miurahr/aqtinstall/pull/846) was merged into master upstream and aqtinstall will soon release its version [3.2.0](https://github.com/miurahr/aqtinstall/discussions/877) including those changes. For it to work, I updated the default value of `aqtversion` to `3.2.*`. I also updated the README.md to use the only Qt LTS remaining (6.8.1) as default version. My temporary version of the Action is available [here]. Note the version `v4.2.1` will only work once the release is published (I will update once it is the case). Until then you must still use `v4.2.0`.

xavier2k6 commented:
`py7zr` default could probably be bumped as well to `==0.22.*` to coincide with aqtinstall/dependencies.

https://github.com/miurahr/aqtinstall/blob/master/pyproject.toml#L29

xavier2k6 commented:
BTW: I have PR #267 Open for the `python-version` range, thoughts?

Kidev commented:
> BTW: I have PR #267 Open for the `python-version` range, thoughts?

It is a good idea, I also have updated it in my latest update. It is not yet ready for this PR, but I'm working on getting commercial versions of Qt being able to be installed with this Action too (https://github.com/miurahr/aqtinstall/pull/878 https://github.com/Kidev/install-qt-action)
![image](https://github.com/user-attachments/assets/45cbbc2f-b184-4a3f-8cae-0376f7818444)

Kidev commented:
Added working support for Qt commercial.

You can see an [example using my version of the workflow](https://github.com/Kidev/QtQuick3D-Tools/actions/runs/12982377381/job/36202105138#step:6:57)  
It uses [my version of `install-qt-action`](https://github.com/marketplace/actions/install-qt-edited) while this PR is not merged  
It also relies on [a PR that was accepted but not yet merged into aqtinstall](https://github.com/miurahr/aqtinstall/pull/878)

Kidev commented:
Fixed the issue after the re-sync, and some force push shenanigans to run the workflows on my repo too

Kidev commented:
Update: almost all tests pass on my repo. There are issues with the `windows-2019` runner however. I'm pretty sure it's because of MSVC2019, that is no longer supported by aqtinstall ([see here](https://github.com/miurahr/aqtinstall/pull/846#issuecomment-2550425227) for the discussion with the maintainer and other contributors of aqtinstall).

jdpurcell commented:
@Kidev Will you be changing the YAML syntax (input parameter name) e.g. from `use-commercial: true` to `use-official: true` to match the recent naming change in aqtinstall? I'm just concerned that if this is merged/released now, and then the parameter name changes, that should be a new major version of this action since it breaks backwards compatibility.

Perhaps it's worth finalizing the YAML syntax now (even if it's still passing the old argument names to aqtinstall, since the renames haven't been released yet), or marking this as a draft if you'd rather wait for the next aqtinstall release, so there's no concern with the YAML backwards compatibility.

Kidev commented:
> @Kidev Will you be changing the YAML syntax (input parameter name) e.g. from `use-commercial: true` to `use-official: true` to match the recent naming change in aqtinstall? I'm just concerned that if this is merged/released now, and then the parameter name changes, that should be a new major version of this action since it breaks backwards compatibility.
> 
> Perhaps it's worth finalizing the YAML syntax now (even if it's still passing the old argument names to aqtinstall, since the renames haven't been released yet), or marking this as a draft if you'd rather wait for the next aqtinstall release, so there's no concern with the YAML backwards compatibility.

That is a nice idea indeed, I would not have done that. I'm doing it right now, been quite busy the last 2 weeks or so

jurplel commented:
> > @Kidev Will you be changing the YAML syntax (input parameter name) e.g. from `use-commercial: true` to `use-official: true` to match the recent naming change in aqtinstall? I'm just concerned that if this is merged/released now, and then the parameter name changes, that should be a new major version of this action since it breaks backwards compatibility.
> > Perhaps it's worth finalizing the YAML syntax now (even if it's still passing the old argument names to aqtinstall, since the renames haven't been released yet), or marking this as a draft if you'd rather wait for the next aqtinstall release, so there's no concern with the YAML backwards compatibility.
> 
> That is a nice idea indeed, I would not have done that. I'm doing it right now along with the changes in commands to aqtinstall, been quite busy the last 2 weeks or so

This sounds good to me—after CI is passing (which I believe it should do if you merge/rebase off of main), we should be good to merge this, then merge @jdpurcell's ARM64 hosts PR

Kidev commented:
This should be good, I'm testing on [my repo](https://github.com/Kidev/QtQuick3D-Tools) as usual. Same as last time, it works on the official repo since my PR on aqtinstall was merged, BUT it is NOT YET RELEASED in a version. So adding this is required until it is the case:
```yaml
aqtsource: 'git+https://github.com/miurahr/aqtinstall.git@master'
```

jdpurcell commented on README.md:
```diff
@@ -74,6 +74,32 @@ When possible, access your Qt directory through the `QT_ROOT_DIR` environment va
+        arch: 'win64_msvc2019_64'
+        aqtsource: 'git+https://github.com/Kidev/aqtinstall.git@install_qt_commercial'
+        use-official: true
+        email: '****@gmail.com'
+        pw: '****'
```
@Kidev Or perhaps (to prevent any unfortunate mistakes 😅):
```yaml
email: ${{ secrets.QT_EMAIL }}
pw: ${{ secrets.QT_PW }}
```

jurplel commented on README.md:
```diff
@@ -74,6 +74,32 @@ When possible, access your Qt directory through the `QT_ROOT_DIR` environment va
+    - name: Install Qt
+      uses: jurplel/install-qt-action@v4
+      with:
+        version: '5.15.3'
```
update this bit of docs with new default settings

jurplel commented:
> This should be good, I'm testing on [my repo](https://github.com/Kidev/QtQuick3D-Tools) as usual. Same as last time, it works on the official repo since my PR on aqtinstall was merged, BUT it is NOT YET RELEASED in a version. So adding this is required until it is the case:
> 
> ```yaml
> aqtsource: 'git+https://github.com/miurahr/aqtinstall.git@master'
> ```

I think we can merge now but hold off on releasing this until the new aqtinstall version drops! I wish we had a way to test this, but I guess since it is a commercial version, there is not really an easy way since we need credentials...

akien-mga commented:
I've been testing these changes to set up Qt with the WASM target (via @Kidev's edited action, version v4.4.3), and I seem to run into an issue with the PATH used for the Linux host. It installs in `qt/$version/gcc_64` but adds `qt/$version/linux_gcc_64/bin` to the `PATH`, so binaries are not in `PATH`.

Here's my config:
```
      - name: Install Qt for Wasm (+ Linux host)
        # Using a fork until https://github.com/jurplel/install-qt-action/pull/272 is merged.
        #uses: jurplel/install-qt-action@v4
        uses: Kidev/install-qt-action@v4.4.3
        with:
          version: 6.8.2
          host: all_os
          target: wasm
          arch: wasm_singlethread
          archives: qtbase qtsvg icu
          modules: qtimageformats
          cache: true
          setup-python: false
          # Depends on a merged PR not yet in latest tagged version.
          aqtsource: git+https://github.com/miurahr/aqtinstall.git@master
          set-env: true
```

```
INFO    : aqtinstall(aqt) v3.2.1.dev11 on Python 3.13.2 [CPython GCC 13.3.0]
/opt/hostedtoolcache/Python/3.13.2/x64/bin/python3 -m aqt install-qt all_os wasm 6.8.2 wasm_singlethread --autodesktop --outputdir /home/runner/work/repo/Qt --modules qtimageformats --archives qtbase qtsvg icu
INFO    : aqtinstall(aqt) v3.2.1.dev11 on Python 3.13.2 [CPython GCC 13.3.0]
INFO    : You are installing the Qt6-WASM version of Qt
...
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/wasm_singlethread/bin/qmake
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/wasm_singlethread/bin/qtpaths
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/wasm_singlethread/bin/qmake6
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/wasm_singlethread/bin/qtpaths6
INFO    : 
INFO    : Autodesktop will now install linux desktop 6.8.2 linux_gcc_64 as required by Qt6-WASM
INFO    : aqtinstall(aqt) v3.2.1.dev11 on Python 3.13.2 [CPython GCC 13.3.0]
...
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/gcc_64/bin/qmake
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/gcc_64/lib/pkgconfig/Qt6Gui.pc
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/gcc_64/lib/pkgconfig/Qt6PrintSupport.pc
INFO    : Patching /home/runner/work/repo/Qt/6.8.2/gcc_64/lib/pkgconfig/Qt6SvgWidgets.pc
...

As you can see at least for Qt 6.8.2, the `linux` `desktop` target gets installed in `gcc_64` and not `linux_gcc_64` (despite the `arch` being named `linux_gcc_64`.

But in the `PATH` set via `set-env: true`, we get:
```
/home/runner/work/repo/Qt/6.8.2/linux_gcc_64/bin
```

---

Another related issue is that all environment variables set by `set-env` are based on the Linux host / `--autodesktop` platform (also with the wrong folder name), and not the `wasm` target which is the main one I'm interested in:

```
QT_ROOT_DIR: /home/runner/work/repo/Qt/6.8.2/linux_gcc_64
QT_PLUGIN_PATH: /home/runner/work/repo/Qt/6.8.2/linux_gcc_64/plugins
QML2_IMPORT_PATH: /home/runner/work/repo/Qt/6.8.2/linux_gcc_64/qml
```

In my case I'd like `QT_ROOT_DIR` and `QT_PLUGIN_PATH` to point to `wasm_singlethread`. But I can imagine other users may have different use cases.

jdpurcell commented:
Yeah, having `QT_ROOT_DIR` point to the target install (not the host install) is the way `install-qt-action` has traditionally worked, so I think we'd want this to function the same way unless there's a good reason not to. I don't think the new WASM-related code in `aqtinstall` is patching `target_qt.conf` in the target install directory, by the way. I mention that because it would be cool if WASM could take advantage of the work I did in #268. Then the entire (I think?) `locateQtWasmHostArchDir` function could go away, `QT_HOST_PATH` would point to the host install, and `QT_ROOT_DIR` would point to the target install.

jdpurcell commented:
I can confirm `QT_HOST_PATH` gets set properly without `locateQtWasmHostArchDir` entirely, as long as `target_qt.conf` is patched, which unfortunately `aqtinstall` is not doing for `wasm`. To test, I made a [small commit](https://github.com/jdpurcell/install-qt-action/commit/46fc3ad415d87ba5af03b5d74d292f397379b233) in my `install-qt-action` fork to enable WASM support. Then I used it to build @Kidev's `QtQuick3D-Tools` in this [workflow run](https://github.com/jdpurcell/QtQuick3D-Tools/actions/runs/13780285087). But rather than `aqtinstall`, I was using [my own Qt installer](https://github.com/jdpurcell/naqt) which patches `target_qt.conf`. Examining the `CMake Build` step, if you expand the first line of the log to see the environment variables, you'll see that both `QT_ROOT_DIR` and `QT_HOST_PATH` are set.

<img width="742" alt="image" src="https://github.com/user-attachments/assets/8e22b976-62b3-4e2c-9414-3b9e9a1666bb" />

Maybe I'll try to fix it in `aqtinstall` if it's something obvious/easy, but honestly I'm tired of dealing with Python code which is why I wrote my own installer.

Kidev commented:
Re, sorry for the delay. I've been having issues running the 'install-qt-official' part on CI. It is very likely due to the very convoluted way it has to be done. Basically its:  
`Python` calls `aqt`, `aqt` installs and calls the `qt-installer` to get data on the packages (`search` command) and then tries to install (`install` command), `qt-installer` does its checks and seems to be setting some env vars for the JWT token then installs. 

This all works well when you have a tty, but Github runners do not. It either fails to auth, or timeouts, and it's very cumbersome to test. I'm still trying though, and for now I'm not sure if it will require changes to `aqt-install`, `install-qt-action`, or both.

I'm going to read your posts and will update accordingly

xavier2k6 commented:
@Kidev Proposed new default of Qt 6.8.1 can be changed to Qt 6.8.3

>(Qt 6.8.3 will be the last public release from Qt 6.8 LTS
* ref.: https://lists.qt-project.org/pipermail/releasing/2025-February/003046.html
* Due for Release ~18th-March-2025
https://wiki.qt.io/Qt_6.8_Release
[Meeting minutes from Qt Release Team meeting 11.03.2025](https://lists.qt-project.org/pipermail/releasing/2025-March/003047.html)

Kidev commented:
`aqtinstall` just released version `3.2.1` containing the official installer. I updated the yml to default to aqt 3.2.1 by default, making it working without needing `aqtsource` anymore

Kidev commented:
Relevant documentation: https://aqtinstall.readthedocs.io/en/latest/official.html

I still need to read your posts, will do

xavier2k6 commented:
**README.md**

>Default: `==3.2.*`

Shouldn't (Default: `==3.2.*`) really state what the proposed default actually is now (with this PR) which would be `3.2.1`?

>aqtversion: '==3.2.*'

Change to (aqtversion: '>=3.2.1') ??

>py7zrversion: '==0.20.*'

Change to (py7zrversion: '==0.22.*') as that's the proposed new default.

jdpurcell commented on README.md:
```diff
@@ -74,6 +74,32 @@ When possible, access your Qt directory through the `QT_ROOT_DIR` environment va
+        version: '5.15.3'
+        target: 'desktop'
+        arch: 'win64_msvc2019_64'
+        aqtsource: 'git+https://github.com/Kidev/aqtinstall.git@install_qt_commercial'
```
```suggestion
```

Kidev commented:
Implemented all recommendations and fixed issues with `QT_HOST_PATH` thanks to @jdpurcell  
Updated the main post as well.

jdpurcell commented:
Looks good! Only last thing I noticed is `aqtversion` changed from `==3.1.*` to ``>=3.2.1``. Perhaps a question for @jurplel but do we want to allow it to automatically use the next minor versions (e.g. 3.3.x) or should it stay locked onto 3.2.x? If the latter, I believe `~=3.2.1` (compatible release clause) would accomplish that.

Kidev commented:
There seems to have been an update on the Qt repos, the name of the installer have been changed. [I pushed a fix in `aqtinstall`](https://github.com/miurahr/aqtinstall/pull/906#issuecomment-2742129652) but in the meantime the official option will fail:  
```
INFO    : Downloading Qt installer to /home/runner/.local/share/aqt/tmp/qt-unified-linux-x64-online.run
ERROR   : Error installing official installer Failed to download installer: Failed to download checksum for the file 'qt-unified-linux-x64-online.run' from mirrors '['https://download.qt.io/']
```

xavier2k6 commented:
>Also Qt 6.8.3 does not exist yet

Got delayed until start of next week -> https://lists.qt-project.org/pipermail/releasing/2025-March/003048.html

Blocker from mailing list has now been resolved -> https://bugreports.qt.io/browse/QTBUG-134289

xavier2k6 commented:
off-topic: any options for FOSS to use Qt 5.15.16/Qt 6.5.5 etc.?

crueter commented:
> off-topic: any options for FOSS to use Qt 5.15.16/Qt 6.5.5 etc.?

KDE maintains these older versions afaict, I'm not sure if they provide anything other than Linux builds however.

xavier2k6 commented:
> > off-topic: any options for FOSS to use Qt 5.15.16/Qt 6.5.5 etc.?
> 
> KDE maintains these older versions afaict, I'm not sure if they provide anything other than Linux builds however.

vcpkg provides latest Qt 5 (Qt 5.15.16) & latest Qt6 (Qt 6.8.x)

Just thought there may be another way via this action to support installing Qt 5.15.16 or indeed Qt 6.5.5 via windows via official installer......not a big deal (no pre-existing compiled binaries available)

crueter commented:
Good opportunity for someone to host binary installers with the KDE builds.

Kidev commented:
> > > off-topic: any options for FOSS to use Qt 5.15.16/Qt 6.5.5 etc.?
> > 
> > 
> > KDE maintains these older versions afaict, I'm not sure if they provide anything other than Linux builds however.
> 
> vcpkg provides latest Qt 5 (Qt 5.15.16) & latest Qt6 (Qt 6.8.x)
> 
> Just thought there may be another way via this action to support installing Qt 5.15.16 or indeed Qt 6.5.5 on windows via official installer......not a big deal (no pre-existing compiled binaries available)

You can install those versions using the official installer, you just need a paid account

xavier2k6 commented:
@Kidev Thanks!

xavier2k6 commented:
Binaries for Qt 6.8.3 are showing now as being available:

https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_683/qt6_683/

xavier2k6 commented:
>**README.md**

>>py7zrversion: '==0.20.*'
>
>Change to (py7zrversion: '==0.22.*') as that's the proposed new default.

Still needs to be changed, Line 311 of this PR

jdpurcell commented:
@Kidev I think I did something like `npm install @actions/cache@4.0.3` but I had a lot of trouble with the way the dependencies are stored in the `node_modules` directory and committed to the repo. That seems to be the old way of doing it and these days the dependencies usually get bundled into a single file. It's also weird because `node_modules` is in `.gitignore` but it shouldn't be when using the older style of publishing where whole directory in the repo. I decided to leave it in `.gitignore` though and update it to do [bundling instead](https://github.com/jdpurcell/install-qt-action/commit/2c2f7d006a30f11efe2d8312f41e08da4bf8b67c).

Kidev commented:
Thanks @xavier2k6 @jdpurcell.
> I think your problem is since `node_modules` is still in `.gitignore`, Git only partially detects changes in there - just modifications and deletions, but not new files added. Even if you remove it from `.gitignore` it's still a pain because then it wants to add a bunch of unnecessary dependencies too (all the `devDependencies`, not just the main ones) unless you are careful when installing/updating dependencies (e.g. `npm install --omit=dev`). I found it easier to do bundling. I'm not much of a Node / JS / TS expert though, that was just my conclusion after a long troubleshooting session with ChatGPT 😄.

Oh that's it for sure, that's a sneaky git error that already happened to me a while back for something else, a sneaky .gitignore mistake. Nice catch! And since I'm not familiar with typescript errors (I use JS but it's for QML aha), I was not able to pinpoint the error because I assumed the whole node_modules dir was pushed ...

Kidev commented:
Everything's done. I used `3.2.*` for `aqtinstall` version so that it works now (except for the `use-official` feature, explained later) by using v3.2.1 but will use v3.2.2 when released. This is important as there were changes on the Qt official installer names, and `aqtinstall` v3.2.1 no longer supports commercial. I already pushed a PR that fixes that, but for it to work we'll need v3.2.2 to be released.   
Used Qt 6.8.3 as LTS in defaults (since it'll be the correct version to use in the long run) however it seems to not yet work for at least WASM, so it kind of holds the PR. You can change it to `6.8.2` if needed.  
Updated @action/cache to v4 to match Github's update

Kidev commented:
I have re-synced with master to limit the python version, so now all checks of the CI pass again ([CI](https://github.com/Kidev/install-qt-action/actions/runs/14366499678) [CI-android](https://github.com/Kidev/install-qt-action/actions/runs/14366499674))
